app/public/js/common/configLocation.js

```diff
@@ -38,12 +38,26 @@ const configuration = {
       userConfigPath = `${userHome}/Library/Preferences/Soundnode`;
     }
 
+    // Guard to assert type of string.
+    if (typeof userConfigPath !== "string") {
+      throw `Could not set userConfigPath for this OS ${process.platform}`
+    }
+
     // create user config in path
     // if there is no userConfig path
-    if (!fs.statSync(userConfigPath).isDirectory()) {
-      this.createUserConfig()
+    //
+    // fs.statSync will throw an exception if
+    // any directory along the path doesn't exist.
+    // Solution: use try catch.
+    try {
```

Comment:
what are other solutions instead `try` `catch`?

Reply:
@weblancaster It appears `stat` throws an error, so we will need to catch the error or rethink the entire functionality of the configLocation